eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/NodeRelay.scala

```diff
@@ -125,18 +129,43 @@ object NodeRelay {
       Some(InvalidOnionPayload(UInt64(2), 0))
     } else {
       payloadOut match {
-        case payloadOut: IntermediatePayload.NodeRelay.Standard =>
-          if (payloadOut.invoiceFeatures.isDefined && payloadOut.paymentSecret.isEmpty) {
-            Some(InvalidOnionPayload(UInt64(8), 0)) // payment secret field is missing
-          } else {
-            None
-          }
-        case _: IntermediatePayload.NodeRelay.ToBlindedPaths =>
-          None
+        // If we're relaying a standard payment to a non-trampoline recipient, we need the payment secret.
+        case payloadOut: IntermediatePayload.NodeRelay.Standard if payloadOut.invoiceFeatures.isDefined && payloadOut.paymentSecret.isEmpty => Some(InvalidOnionPayload(UInt64(8), 0))
+        case _: IntermediatePayload.NodeRelay.Standard => None
+        case _: IntermediatePayload.NodeRelay.ToBlindedPaths => None
       }
     }
   }
 
+  private def isAsyncPayment(nodeParams: NodeParams, nextPayload: IntermediatePayload.NodeRelay): Boolean = {
+    nextPayload match {
+      case payload: IntermediatePayload.NodeRelay.Standard => nodeParams.features.hasFeature(Features.AsyncPaymentPrototype) && payload.isAsyncPayment
+      // We haven't added the async payment flag to the blinded path's encrypted data yet.
+      case _: IntermediatePayload.NodeRelay.ToBlindedPaths => false
+    }
+  }
+
+  private def shouldWakeUpNextNode(nodeParams: NodeParams, recipient: Recipient): Boolean = {
+    recipient match {
+      case r: BlindedRecipient => r.blindedHops.head.resolved.route match {
+        case BlindedPathsResolver.PartialBlindedRoute(_: EncodedNodeId.WithPublicKey.Wallet, _, _) => true
+        case _ => false
+      }
+      case _ => false
```

Comment:
This case can be extended to wake-up current Phoenix wallets: if this is a trampoline->trampoline payment, we should check our push notifications DB here and if it has an entry for the next node, we should return `true`.